worker/create_batches_from_filters_worker.go

```diff
@@ -23,265 +23,137 @@
-	b.PushDB, err = extensions.NewPGClient("push.db", b.Config, b.Logger)
-	checkErr(b.Logger, err)
-}
-
-func (b *CreateBatchesFromFiltersWorker) configureMarathonDatabase() {
-	var err error
-	b.MarathonDB, err = extensions.NewPGClient("db", b.Config, b.Logger)
-	checkErr(b.Logger, err)
-}
-
-func (b *CreateBatchesFromFiltersWorker) loadConfigurationDefaults() {
-	b.Config.SetDefault("workers.createBatchesFromFilters.dbPageSize", 1000)
-	b.Config.SetDefault("workers.createBatchesFromFilters.pageProcessingConcurrency", 1)
-}
-
-func (b *CreateBatchesFromFiltersWorker) loadConfiguration() {
-	b.DBPageSize = b.Config.GetInt("workers.createBatchesFromFilters.dbPageSize")
-	b.PageProcessingConcurrency = b.Config.GetInt("workers.createBatchesFromFilters.pageProcessingConcurrency")
-}
-
-func (b *CreateBatchesFromFiltersWorker) configureDatabases() {
-	b.configureMarathonDatabase()
-	b.configurePushDatabase()
-}
-
-func (b *CreateBatchesFromFiltersWorker) configureRedisClient() {
-	r, err := extensions.NewRedis("workers", b.Config, b.Logger)
-	checkErr(b.Logger, err)
-	b.RedisClient = r
-}
-
-func (b *CreateBatchesFromFiltersWorker) configureS3Client() {
-	s3Client, err := extensions.NewS3(b.Config, b.Logger)
-	checkErr(b.Logger, err)
-	b.S3Client = s3Client
-}
-
-func (b *CreateBatchesFromFiltersWorker) configure() {
-	b.loadConfigurationDefaults()
-	b.loadConfiguration()
-	b.configureDatabases()
-	b.configureS3Client()
-	b.configureRedisClient()
-}
-
-func (b *CreateBatchesFromFiltersWorker) getPageFromDBWithFilters(job *model.Job, page DBPage) *[]User {
+func (b *CreateBatchesFromFiltersWorker) createDbToCsvJob(job *model.Job, page DBPage,
+	uploader *s3.CreateMultipartUploadOutput, total int) {
 	filters := job.Filters
 	whereClause := GetWhereClauseFromFilters(filters)
-	limit := b.DBPageSize
 	var query string
 	if (whereClause) != "" {
-		query = fmt.Sprintf("SELECT user_id FROM %s WHERE seq_id > %d AND %s ORDER BY seq_id ASC LIMIT %d;", GetPushDBTableName(job.App.Name, job.Service), page.Offset, whereClause, limit)
+		query = fmt.Sprintf("SELECT DISTINCT user_id FROM %s WHERE user_id > '%s' AND user_id <= '%s' AND %s ORDER BY user_id;", GetPushDBTableName(job.App.Name, job.Service), page.SmallestID, page.BiggestID, whereClause)
 	} else {
-		query = fmt.Sprintf("SELECT user_id FROM %s WHERE seq_id > %d ORDER BY seq_id ASC LIMIT %d;", GetPushDBTableName(job.App.Name, job.Service), page.Offset, limit)
+		query = fmt.Sprintf("SELECT DISTINCT user_id FROM %s WHERE user_id > '%s' AND user_id <= '%s' ORDER BY user_id;", GetPushDBTableName(job.App.Name, job.Service), page.SmallestID, page.BiggestID)
 	}
-	var users []User
-	start := time.Now()
-	_, err := b.PushDB.DB.Query(&users, query)
-	labels := []string{fmt.Sprintf("game:%s", job.App.Name), fmt.Sprintf("platform:%s", job.Service)}
-	b.Workers.Statsd.Timing("get_page_with_filters", time.Now().Sub(start), labels, 1)
-
-	checkErr(b.Logger, err)
-	return &users
+	b.Workers.DbToCsvJob(&ToCSVMenssage{
```

Comment:
r/Menssage/Message

I suggest you do a find/replace for Menssage in your code :)